include/irr/asset/ICPUVirtualTexture.h

```diff
@@ -55,70 +55,109 @@ class ICPUVirtualTexture final : public IVirtualTexture<ICPUImageView>
         }
     };
 
+    static auto createPoTPaddedSquareImageWithMipLevels(const asset::ICPUImage* _img, asset::ISampler::E_TEXTURE_CLAMP _wrapu, asset::ISampler::E_TEXTURE_CLAMP _wrapv)
+    {
+        const auto& params = _img->getCreationParameters();
+        const auto originalExtent = params.extent;
+        const uint32_t paddedExtent = core::roundUpToPoT(std::max(params.extent.width,params.extent.height));
+
+        //create PoT and square image with regions for all mips
+        asset::ICPUImage::SCreationParams paddedParams = params;
+        paddedParams.extent = {paddedExtent,paddedExtent,1u};
+        //in case of original extent being non-PoT, padding it to PoT gives us one extra not needed mip level (making sure to not cumpute it)
```

Comment:
that would be correct, a padded texture has one extra mip, example

a 65 wide texture with mip-map rounding UP
65,33,17,9,5,3,2,1 = 8 mips

a 128 wide padded texture 
128,64,32,16,8,4,2,1 = 8 mips